File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/admin/user/contoller/AdminUserManageController.java
package com.multicampus.gamesungcoding.a11ymarketserver.admin.user.contoller;

import com.multicampus.gamesungcoding.a11ymarketserver.admin.user.service.AdminUserManageService;
import com.multicampus.gamesungcoding.a11ymarketserver.user.model.User;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@Slf4j
@RestController
@RequiredArgsConstructor
public class AdminUserManageController {
    private final AdminUserManageService userService;

    @GetMapping("/v1/admin/users")
    public List<String> inquireUsers(@RequestParam String userId) {
        log.info("inquireUsers userId={}", userId);

        return userService.listAll().stream().map(User::getUserName).toList();
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/admin/user/controller/AdminUserManageController.kt
package com.multicampus.gamesungcoding.a11ymarketserver.admin.user.controller

import com.multicampus.gamesungcoding.a11ymarketserver.admin.user.service.AdminUserManageService
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto.UserAdminResponse
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto.UserResponse
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserRole
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/api")
class AdminUserManageController(
    private val userService: AdminUserManageService
) {
    // 관리자 - 전체 사용자 조회
    @GetMapping("/v1/admin/users")
    fun inquireUsers(): List<UserAdminResponse> =
        userService.listAll()

    // 관리자 - 사용자 권한 변경
    @PatchMapping("/v1/admin/users/{userId}/permission")
    fun changeUserPermission(
        @PathVariable userId: UUID,
        @RequestParam role: UserRole
    ): UserResponse = userService.changePermission(userId, role)
}
